File: mobile-sdks/ios/YextAnalytics/Sources/YextAnalytics/YextAnalytics.swift
import Foundation

/// Main entry point for the Yext Analytics Events SDK
/// Returns an AnalyticsEventService given an AnalyticsConfig
public func analytics(config: AnalyticsConfig) throws -> AnalyticsEventService {
    return try AnalyticsEventReporter(config: config)
}

// Export all public types
public typealias YextAnalyticsConfig = AnalyticsConfig
public typealias YextEventPayload = EventPayload
public typealias YextAction = Action
public typealias YextAnalyticsEventService = AnalyticsEventService
public typealias YextAnalyticsError = AnalyticsError

File: mobile-sdks/android/yext-analytics/src/main/java/com/yext/analytics/YextAnalytics.kt
package com.yext.analytics

import android.content.Context

/**
 * Main entry point for the Yext Analytics Events SDK
 * Returns an AnalyticsEventService given an AnalyticsConfig and Android Context
 */
object YextAnalytics {
    /**
     * Create an analytics event service with the given configuration
     */
    fun analytics(config: AnalyticsConfig, context: Context): AnalyticsEventService {
        return AnalyticsEventReporter.create(config, context)
    }
}